pkg/storage/streams/store.go

```diff
@@ -84,19 +94,31 @@ func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	var lastSegmentSize int64
 
 	awareLimitReader := EOFAwareReader(data)
+	encKey := sha256.Sum256([]byte(s.key))
+	var firstNonce [12]byte
 
 	for !awareLimitReader.isEOF() {
+		// TODO (moby) should I create a new encrypter for each segment?
```

Comment:
eventually, yes. for now, maybe not?

Reply:
if you are going to do it once per stream, then you should set up the Transformer prior to even creating the EOFAwareLimitReader. right now, this is definitely encrypting different segments with the same key and same starting nonce, which is a security hazard.

Reply:
@jtolds is it really encrypting all segments with the same starting nonce? `firstNonce` is initialized outside the for-loop and passed as a pointer to the encrypter. I would expect that `firstNonce` is zero only for the first segment and then incremented to something by the encrypter on each iteration. Am I missing anything?